lib/__tests__.js

```diff
@@ -1151,112 +1150,151 @@ Test('federated schema can include custom directive', (t) => {
-    st.equal(customMakeExecutableSchema.calledOnce, true, 'calls custom makeExecutableSchema');
-  });
-
-  t.test('component schema uses custom makeExecutableSchema when federation is specified but undefined', async (st) => {
-    st.plan(2);
-    const customMakeExecutableSchema = sinon.spy();
-    const componentA = new GraphQLComponent({
-      ...baseComponentOptions,
-      makeExecutableSchema: customMakeExecutableSchema,
-      federation: undefined
+  
+    const rootComponent = new GraphQLComponent({
+      imports: [component1, component2],
+      pruneSchema: true
     });
-
-    st.doesNotThrow(() => componentA.schema, 'builds schema correctly');
-    st.equal(customMakeExecutableSchema.calledOnce, true, 'calls custom makeExecutableSchema');
+  
+    const { schema } = rootComponent;
+  
+    console.log(graphql.printSchema(schema));
+  
+    t.ok(schema.getType('Property'), 'Property type exists');
+    t.ok(schema.getType('Review'), 'Review type exists');
+  
+    t.end();
   });
 
-  t.test('component schema uses custom makeExecutableSchema when federation is specified but null', async (st) => {
-    st.plan(2);
-    const customMakeExecutableSchema = sinon.spy();
-    const componentA = new GraphQLComponent({
-      ...baseComponentOptions,
-      makeExecutableSchema: customMakeExecutableSchema,
-      federation: null
-    });
-
-    st.doesNotThrow(() => componentA.schema, 'builds schema correctly');
-    st.equal(customMakeExecutableSchema.calledOnce, true, 'calls custom makeExecutableSchema');
-  });
-
-  t.test('component schema does not use custom makeExecutableSchema when specified but undefined', async (st) => {
-    st.plan(1);
-    const componentA = new GraphQLComponent({
-      ...baseComponentOptions,
-      makeExecutableSchema: undefined
-    });
-    st.doesNotThrow(() => componentA.schema, 'builds schema correctly');
-  });
+  t.test('Unused types not removed if prune is false', (t) => {
+    
+    const component = new GraphQLComponent({
+      types: `
+        type Query {
+          reviews(propertyId: ID!): [Review]
+        }
+  
+        type Review {
+          id: ID!
+          content: String
+        }
 
-  t.test('component schema does not use custom makeExecutableSchema when specified but null', async (st) => {
-    st.plan(1);
-    const componentA = new GraphQLComponent({
-      ...baseComponentOptions,
-      makeExecutableSchema: null
+        type Property {
+          reviews: [Review]
+        }
+      `,
+      resolvers: {
+        Query: {
+          reviews() {
+            return {
+              id: 'rev-id-1',
+              content: 'content'
+            }
+          }
+        },
+      },
+      pruneSchema: false
     });
-    st.doesNotThrow(() => componentA.schema, 'builds schema correctly');
+  
+    const { schema } = component;
+  
+    console.log(graphql.printSchema(schema));
+  
+    t.ok(schema.getType('Property'), 'Property type was pruned');
```

Comment:
doesn't this assertion prove that this test is not working as intended? The test is titled: "Unused types not removed if prune is false" yet the `Property` type was removed as per the assertion.

Reply:
+1, we need to change the assertion message to `Property type was not pruned` or something like that as in this case we are setting `pruneSchema: false`

Reply:
Just a typo